Include projectDir in the APK path samples

Sloy opened the pull request:
Hi! 
I was trying to create a custom task to run a subset of test, and I just spent a couple of hours trying to figure out why the plugin wasn't finding the APKs. So I decided to open this PR in case you think it would be useful for other people in the future. Also added a new test case, which I used to debug the issue.

### Short version:
When the `File` objects pointing to the APKs are created, they're relativ. Because of that Composer would not find them. No matter how I wrote them, either the plugin could not find it, or it could but then the Composer command would not. Weird. Although running the task from AndroidStudio would work. Even weirder. In the end I found out that adding `${projectDir}` as prefix to the path would make it work in all cases. ¯\\\_(ツ)\_/¯

### Very long version
Turns out the `File` object pointing to the APK created in both `ConfiguratorDomainObj` and `ComposerTask` with 
```
override fun apk(value: String) {
    apk(File(value))
}
```
creates a relative path. 

Because of this, I saw that execution would be inconsistent depending on how you write the path and where you ran the task from. 

For example in the test I added, when using `apk "build/outputs/...` the plugin would find the apk but then Composer would use a wrong path like `/Users/rafael.vazquez/Android/composer-gradle-plugin/plugin/build/outputs/...` and fail. (Should be using the temporary folder like `/private/var/folders/r9/lhsyb_795qgdv0524gdyfyyw0000gp/T/junit8993139078181683454/vanilla/vanilla/build/outputs/...`.

So, I tried to make the file path absolute. It can be done inside `ComposerTask` without any problem with
```
override fun apk(value: String) {
    apk(File(project.projectDir, value))
}
```

But I didn't find a way to do the same in `ConfiguratorDomainObj` because this object doesn't have direct access to the `project` instance. Maybe there is a way but I could not find one 😢 

So I went for the fast and cheap solution, just use the full path in the gradle configuration 😂 

I hope this is useful!
Feel free to reject any or all of the changes if you don't think it's the best approach.

And thanks for the plugin, the neat code and those lovely integration tests (they helped me a lot to find the issue) 😃

Sloy committed:
Add test to verify the custom task can find the APKs

Depending on how you write the path to the APK, the plugin might not be able to find it.
I added a test to verify the successful scenario.

Sloy committed:
Include projectDir in the APK path samples

Without it, the plugin will use a relative path that will change depending on where you run the task from.
Doing it this way ensures that the path will always be the right one.

trevjonez commented:
The existing `ComposerConfigurator` should be able to handle your use case as is. it exposes both the apk and testApk settings with a function that accepts a `File` as well as `String` which means that rather than `apk "${projectDir}/path/to.apk" ` you can use `apk project.file("path/to.apk")`. 

That said I think what I want to do is tweak the readme a bit more and modify the way we handle strings passed to the `ComposerTask`. 

Thanks for bringing this to my attention. oversight on my part for sure.

trevjonez committed:
readme update to notate how string paths are resolved
readme update to notate how File objects can be used
lazy task registration/eval
latest composer
latest kotlin
latest gradle wrapper

trevjonez commented:
ok so slightly different than what you had originally proposed but should be best long term solution by depending on project.file(...). 

available via [0.7.0](https://github.com/trevjonez/composer-gradle-plugin/releases/tag/0.7.0)

Sloy commented:
Ah! I didn't think you could also pass a file object to the configuration! Even though I saw the method, I thought it was an alias for internal use.
Nice addition to the readme, would have saved me a lot of time. I hope it helps people in the future :)
